Fix DateRange, SizeRange, FromAddresses and SentToAddresses references.

cduarte-ta opened the pull request:
Fixes the retrieval of a few attributes by matching the xml specifications.

ecederstrand commented:
Thanks for the contribution!

https://docs.microsoft.com/en-us/exchange/client-developer/web-service-reference/withindaterange and https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/withinsizerange state that they are in the Messages namespace (MNS). Is that not true in practice?

cduarte-ta commented:
Hi Erik. Unfortunately I don't have access to an exchange server any more. At the time I was doing a converter from outlook rules to gmail equivalent filters and those fields were coming empty. After the patch I was able to retrieve them.

```
# note, conditions: exchangelib.properties.Conditions
        if (o := conditions.within_date_range) is not None:
            date_range : exchangelib.properties.WithinDateRange = o
            start : exchangelib.EWSDateTime = date_range.start_date_time
            if start is not None:
                # transform 2009-01-15T13:45:56Z into 2009/01/15 which is accepted by gmail
                # ignoring any timezone info
                x = start.ewsformat().split("T")[0].replace("-", "/")
                self._add_query(out_filters, [x], "after:")
            end : exchangelib.EWSDateTime = date_range.end_date_time
            if end is not None:
                x = end.ewsformat().split("T")[0].replace("-", "/")
                self._add_query(out_filters, [x], "before:")

        if (o := conditions.within_size_range) is not None:
            size_range : exchangelib.properties.WithinSizeRange = o
            if size_range.minimum_size is not None:
                self._add_query(out_filters, [o.minimum_size * 1024], "larger:")
            if size_range.maximum_size is not None:
                self._add_query(out_filters, [o.maximum_size * 1024], "smaller:")
```

cduarte-ta commented:
The c# library is writing those elements in the types namespace, c.f. 
https://github.com/OfficeDev/ews-managed-api/blob/master/ComplexProperties/RulePredicateDateRange.cs#L111

ecederstrand commented:
Thanks for the extra info. Appreciated!
